templates/client/html/catalog/detail/body.php

```diff
@@ -620,3 +620,30 @@
 	</div>
 
 <?php endif ?>
+
+<?php
+// preselection of a variant in a selection article
+if (isset($_GET['productId']) && is_numeric($_GET['productId'])):
+
+    foreach( $this->detailProductItem->getRefItems( 'product', null, 'default' ) as $prodid => $product ) :
+
+        if ($prodid == $_GET['productId']):
+
+            $attributes = $product->getRefItems('attribute',null,'variant');
+            $js = '';
+            foreach($attributes as $attribId => $attribute):
+                $js .= '"' .  $attribute->getType() . '": "' . $attribute->getId() . '",
+                ';
+            endforeach;
+
+            if (!empty($js)): ?>
+               <script>
+                   AimeosVariantSelectionProduct = {"prodId": "<?= $this->detailProductItem->getId() ?>", "attributes": {<?= $js ?>}};
+               </script>
```

Comment:
Adding Javascript in the template won't work if Content-Security-Policy is enabled, which is done by default in most installations. You have to add that to a "data-..." attribute of an element instead.